server/modules/instantiation/port-provider.js

```diff
@@ -52,28 +57,31 @@ module.exports = function () {
         }
 
         providePort(expectedPortRanges) {
-            var i, port;
-            for (i = 0; i < expectedPortRanges.length; i += 1) {
-                for (port = expectedPortRanges[i].minPort; port <= expectedPortRanges[i].maxPort; port += 1) {
-                    if (this.availablePortRanges.contains(port) && !this.externalPortRanges.contains(port)) {
-                        this.addExternalPortRange(port, port);
-
-                        return port;
-                    }
-                }
-            }
-
-            throw new Error('Failed to allocate port.');
+            return Promise
+                .any(expectedPortRanges.map(portRange => this.findPort(portRange)))
```

Comment:
initially I thought so as well
this actually was the biggest struggle here and after all we found out this changes the context and `this` becomes something else than it used to be